poc_mobile_verifier/src/verifier.rs

```diff
@@ -0,0 +1,197 @@
+use std::ops::Range;
+
+use crate::{
+    error::{Error, Result},
+    heartbeats::{Heartbeat, Heartbeats},
+    shares::Shares,
+    subnetwork_rewards::SubnetworkRewards,
+};
+use chrono::{DateTime, Duration, TimeZone, Utc};
+use db_store::MetaValue;
+use file_store::{file_sink, FileStore};
+use helium_proto::services::{follower, Channel};
+use sqlx::{Pool, Postgres};
+use tokio::time::sleep;
+
+pub struct VerifierDaemon {
+    pub pool: Pool<Postgres>,
+    pub valid_shares_tx: file_sink::MessageSender,
+    pub invalid_shares_tx: file_sink::MessageSender,
+    pub subnet_rewards_tx: file_sink::MessageSender,
+    pub reward_period_hours: i64,
+    pub verifications_per_period: i32,
+    pub verifier: Verifier,
+}
+
+impl VerifierDaemon {
+    pub async fn run(mut self, shutdown: &triggered::Listener) -> Result {
+        tracing::info!("Starting verifier service");
+
+        let reward_period = Duration::hours(self.reward_period_hours);
+        let verification_period = reward_period / self.verifications_per_period;
+
+        loop {
+            let now = Utc::now();
+            // Maybe name these "epoch_since_*" and "epoch_since_*_duration"
```

Comment:
i like this; `get_*_epoch` implies a beginning and end and both of those aren't clear when you only have `now` as the frame of reference; "is it from some point in the past until now or is it from now until some point in the future?"